File: src/e_149/Disemvoweler.java
package e_149;

/**
 * Created by osk on 2016-02-21.
 */
public class Disemvoweler {
    private static final String[] VOWELS = { "a", "o", "u", "e", "i", "u" };

    public static void main(String[] args) {
        String text = "did you hear about the excellent farmer who was outstanding in his field";
        disemvowel(text);
    }

    private static void disemvowel(String text) {
        String result = "";
        String vowel = "";

        for (int i = 0; i < text.length(); i++) {
            if (contains(VOWELS, String.valueOf(text.charAt(i)))) {
                vowel += String.valueOf(text.charAt(i));
            } else if (text.charAt(i) != ' ') {
                result += String.valueOf(text.charAt(i));
            }
        }
        System.out.println(result);
        System.out.println(vowel);
    }

    private static boolean contains(String[] array, String element) {
        for (String s : array) {
            if (s.equals(element)) {
                return true;
            }
        }
        return false;
    }

}

File: src/149e/disemvoweler.cc
#include <iostream>
#include <vector>
#include <string>
#include <ctime>
#include <algorithm>

using namespace std;

vector<string> VOWELS = { "a", "o", "u", "e", "i", "u" };

void disemvowel(string text) {
	string result;
	string vowel;

	for (int i = 0; i < text.length(); i++) {
		if (std::find(VOWELS.begin(), VOWELS.end(), string(1, text[i])) != VOWELS.end()) {
			vowel += text[i];
		} else if (text[i] != ' ') {
			result += text[i];
		}
	}
	std::cout << result << "\n" << vowel << endl;
}

int main() {
	string text = "did you hear about the excellent farmer who was outstanding in his field";
	disemvowel(text);
}